CHANGELOG.md

```diff
@@ -42,6 +42,8 @@ Note that 3.1.0 will be the first API stable release and interfaces in this rele
 * We lowered the requirements of the `seqan3::aligned_sequence` concept by removing everything that needs write
   access to the object. We then added a new `seqan3::writable_aligned_sequence` concept which extends
   `seqan3::aligned_sequence` by the requirements that need write access (e.g. `insert_gap`).
+* We put some `align_cfg` elements into a subnamespace `align_cfg::detail`:
+  `align_cfg::detail::debug` ([\#2020](https://github.com/seqan/seqan3/pull/2020)).
```

Comment:
@seqan/core Do we want to add a CHANGELOG entry for moved entities from `seqan3::align_cfg` namespace to `seqan3::align_cfg::detail`?

Reply:
hm should the user know about the `debug` functionality at all? If not, I wouldn't put it into the changelog

Reply:
Yesterday I discussed this with @joergi-w because he uses many of our detailed functions for his app. The debug_stream, for example, is very practical.

Reply:
The debug_stream is not detail, it's public API.

https://github.com/seqan/seqan3/blob/c24c911880d350e3388616563ad73b9ca9ac0ba7/include/seqan3/alignment/configuration/align_config_debug.hpp#L19-L38

`seqan3::align_cfg::debug` is semi-public and only used within tests. If it is the only change, I wouldn't mention it. (I only cursorily looked over the PR)

Reply:
No the debug thing is developer API only. And as far as I know it was also only documented like so. Which means the regular user does not know about it. That's why we decided to put it into a specific detail sub namespace.

> Yesterday I discussed this with @joergi-w because he uses many of our detailed functions for his app. The debug_stream, for example, is very practical.

That is the responsibility of the app developer. We don't give any guarantees about the detail namespace. That can change at any time without notice.